core/src/main/java/com/graphhopper/routing/util/parsers/OSMMaxSpeedParser.java

```diff
@@ -54,11 +53,26 @@ public void createEncodedValues(EncodedValueLookup lookup, List<EncodedValue> li
 
     @Override
     public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, boolean ferry, IntsRef relationFlags) {
+        return handleWayTags(edgeFlags, way, ferry, relationFlags, Collections.emptyList());
+    }
+
+    @Override
+    public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, boolean ferry, IntsRef relationFlags, List<CustomArea> customAreas) {
         double maxSpeed = OSMValueExtractor.stringToKmh(way.getTag("maxspeed"));
 
+        RoadClass roadClass = RoadClass.find(way.getTag("highway", ""));
+        for (CustomArea customArea : customAreas) {
+            Object countryCode = customArea.getProperties().get("ISO3166-1:alpha3");
+            if (countryCode != null) {
+                NewCountry country = NewCountry.valueOf(countryCode.toString());
+                CountryRule countryRule = CountryRule.getCountryRule(country);
+                if (countryRule != null) {
+                    maxSpeed = countryRule.getMaxSpeed(roadClass, TransportationMode.CAR, maxSpeed);
+                }
+            }
+        }
```

Comment:
We can move most of this into OSMReader and pass the country as an artificial way tag instead.